scope_registry_test.go

```diff
@@ -140,3 +141,21 @@ func TestNewTestStatsReporterManyScopes(t *testing.T) {
 		wantHistograms, r.counters[histogramCardinalityName].val,
 	)
 }
+
+func TestForEachScope(t *testing.T) {
+	scope := NewTestScope("", nil)
+	go func() {
+		for {
+			hello := scope.Tagged(map[string]string{"a": "b"}).Counter("hello")
+			hello.Inc(1)
+		}
+	}()
+	var val CounterSnapshot
+	for {
+		val = scope.Snapshot().Counters()["hello+a=b"]
```

Comment:
This is kind of a nit, but should we express the test in terms of `ForEachScope` instead? Since that's where the actual bug is. Could do both tests if we're paranoid haha.

Reply:
done, though ForEachScope()'s only real use case is snapshot() so the tests are pretty much the same